hoot-core/src/main/cpp/hoot/core/elements/Tags.cpp

```diff
@@ -166,6 +167,159 @@ double Tags::getDouble(const QString& k, double defaultValue) const
   }
 }
 
+double Tags::getVelocity(const QString& k) const
+{
+  QString v = get(k);
+
+  bool isKm = false;    //kilometer per hour
+  bool isMph = false;   //miles per hour
+  bool isKnots = false; //knots
+  if (v.contains("km/h", Qt::CaseInsensitive) || v.contains("kph", Qt::CaseInsensitive)
+      || v.contains("kmph", Qt::CaseInsensitive))
+  {
+     isKm = true;
+  }
+  else if (v.contains("mph", Qt::CaseInsensitive))
+  {
+    isMph = true;
+  }
+  else if (v.contains("knots", Qt::CaseInsensitive))
+  {
+    isKnots = true;
+  }
+  else //default kilometre per hour is assumed
+  {
+    isKm = true;
+  }
+
+  bool ok;
+  double result = v.split(" ")[0].toDouble(&ok);
```

Comment:
It looks like the units aren't always separated by a space. E.g. "30mph" or "20kph".

See the second page here: http://taginfo.openstreetmap.org/keys/maxspeed#values